src/nanovg_pick.c

```diff
@@ -0,0 +1,1742 @@
+		}
+		else
+		{
+			nvg__bezierBounds(points, seg->bounds);
+		}
+	}
+}
+
+static void nvg__pickSubPathAddStrokeSupports(NVGpickScene* ps, NVGpickSubPath* psp, float strokeWidth, int lineCap, int lineJoin, float miterLimit)
+{
+	int closed = psp->closed;
+	const float* points = ps->points;
+
+	NVGsegment* prevseg = NULL;
+	NVGsegment* seg = NULL;
+	NVGsegment* segments = &ps->segments[psp->firstSegment];
+	int nsegments = psp->nsegments;
+	
+	prevseg = psp->closed ? &segments[psp->nsegments - 1] : NULL;
+	
+	int ns = 0; // nsupports
+	float supportingPoints[32];
+	int firstPoint, lastPoint;
+
+	if (closed == 0)
+	{
+		segments[0].flags |= NVG_PICK_CAP;
+		segments[nsegments - 1].flags |= NVG_PICK_ENDCAP;
+	}
+	
+	int s;	
+	for (s = 0; s < nsegments; ++s)
+	{
+		seg = &segments[s];
+		nvg__initBounds(seg->bounds);
+		
+		firstPoint = seg->firstPoint * 2;
+		lastPoint = firstPoint + ((seg->type == NVG_LINETO) ? 2 : 6);
+		
+		ns = 0;
+		
+		// First two supporting points are either side of the start point
+		supportingPoints[ns++] = points[firstPoint] - seg->startDir[1] * strokeWidth;
+		supportingPoints[ns++] = points[firstPoint+1] + seg->startDir[0] * strokeWidth;
+	
+		supportingPoints[ns++] = points[firstPoint] + seg->startDir[1] * strokeWidth;
+		supportingPoints[ns++] = points[firstPoint+1] - seg->startDir[0] * strokeWidth;
+
+		// Second two supporting points are either side of the end point
+		supportingPoints[ns++] = points[lastPoint] - seg->endDir[1] * strokeWidth;
+		supportingPoints[ns++] = points[lastPoint+1] + seg->endDir[0] * strokeWidth;
+	
+		supportingPoints[ns++] = points[lastPoint] + seg->endDir[1] * strokeWidth;
+		supportingPoints[ns++] = points[lastPoint+1] - seg->endDir[0] * strokeWidth;
+
+//		NVG_PICK_DEBUG_VECTOR_SCALE(&points[firstPoint], seg->startDir, 10);
+//		NVG_PICK_DEBUG_LINE(&points[firstPoint], &points[lastPoint]);
+//		NVG_PICK_DEBUG_VECTOR_SCALE(&points[lastPoint], seg->endDir, 10);
+		
+		if (seg->flags & NVG_PICK_CORNER && prevseg != NULL)
+		{
+			seg->miterDir[0] = 0.5f * (-prevseg->endDir[1] - seg->startDir[1]);
+			seg->miterDir[1] = 0.5f * (prevseg->endDir[0] + seg->startDir[0]);
+
+			float M2 = 		seg->miterDir[0] * seg->miterDir[0]
+						+	seg->miterDir[1] * seg->miterDir[1];
+						
+			if (M2 > 0.000001f) {
+				float scale = 1.0f / M2;
+				if (scale > 600.0f) {
+					scale = 600.0f;
+				}
+				seg->miterDir[0] *= scale;
+				seg->miterDir[1] *= scale;
+			}
+
+			NVG_PICK_DEBUG_VECTOR_SCALE(&points[firstPoint], seg->miterDir, 10);
+
+			// Add an additional support at the corner on the other line
+			supportingPoints[ns++] = points[firstPoint] - prevseg->endDir[1] * strokeWidth;
+			supportingPoints[ns++] = points[firstPoint+1] + prevseg->endDir[0] * strokeWidth;
+
+			if (lineJoin == NVG_MITER || lineJoin == NVG_BEVEL)
+			{							
+				// Set a corner as beveled if the join type is bevel or mitered and 
+				// miterLimit is hit.
+				if (	lineJoin == NVG_BEVEL
```

Comment:
Same line, for multiliners, operator at end, just one tab indetation for next row, curly after if